app/views/AnalyticalFramework/components/Canvas/index.tsx

```diff
@@ -326,33 +327,47 @@ function Canvas<T>(props: Props<T>) {
             </>
         );
     }
+
     return (
-        <>
-            <NonFieldError
-                error={error}
-            />
-            <SortableList
-                className={styles.canvas}
-                name="widgets"
-                onChange={handleWidgetOrderChangeComplete}
-                // eslint-disable-next-line react/destructuring-assignment
-                data={props.widgets}
-                keySelector={widgetKeySelector}
-                renderer={AttributeInputWrapper}
-                direction="rect"
-                rendererParams={widgetRendererParams}
-                itemContainerParams={itemContainerParams}
-                showDragOverlay
-                emptyIcon={(
-                    <Kraken
-                        variant="sleep"
-                    />
-                )}
-                emptyMessage="No widgets found."
-                messageShown
-                messageIconShown
-            />
-        </>
+        props.widgets && props.widgets.length > 0 ? (
+            <>
+                <NonFieldError
+                    error={error}
+                />
+                <SortableList
+                    className={styles.canvas}
+                    name="widgets"
+                    onChange={handleWidgetOrderChangeComplete}
+                    // eslint-disable-next-line react/destructuring-assignment
+                    data={props.widgets}
+                    keySelector={widgetKeySelector}
+                    renderer={AttributeInputWrapper}
+                    direction="rect"
+                    rendererParams={widgetRendererParams}
+                    itemContainerParams={itemContainerParams}
+                    showDragOverlay
+                    emptyIcon={(
+                        <Kraken
+                            variant="sleep"
+                        />
+                    )}
+                    emptyMessage="No widgets list found."
+                    messageShown
+                    messageIconShown
+                />
+            </>
+        ) : (
+            <div className={styles.noWidgetsFound}>
+                <Message
+                    icon={(
+                        <Kraken
+                            variant="sleep"
+                        />
+                    )}
+                    message="No widgets found."
+                />
+            </div>
+        )
```

Comment:
1. We already have ListView to handle empty messages here.
2. Why change emptyMessage on ListView

Reply:
The mentioned issue required that the empty message should appear at the center of the component and to fix this issue a separate component hiding the layer of was used. (ref.1) 
For (ref.2) the emptyMessage has been changed to "No widgets list found."